### DISCUSSION TITLE
PaperScroller disable scrollbar and sizing issues

### DISCUSSION THREAD

**Discussion by Anuthama-Ganesan** [clientIO/joint#1829] [Q&A] [answered]
I have added a paperscroller to my paper and have given the height and width for it. Below shows my code snippet. 
```
 const graph = new joint.dia.Graph();
    const paper = new joint.dia.Paper
      ({
        model: graph,
        background: {
          color: '#202020',
        },
        width: '100%',
        height: '700px',
        frozen: true,
        async: true,
        sorting: joint.dia.Paper.sorting.EXACT,
        cellViewNamespace: joint.shapes,
        gridSize: 20,
        perpendicularLinks: true,
      });

    const scroller = new joint.ui.PaperScroller({
      paper,
      autoResizePaper: true,
      scrollWhileDragging: true,
      baseWidth: 100,
      baseHeight: 50,
      cursor: 'grab',
      contentOptions: {
        minWidth: 100,
        minHeight: 50,
      }
    });

    canvas.current.appendChild(scroller.el);
    scroller.render().center();
    paper.fitToContent();
```
Even after providing the height and width, I see the style to not be included in the `joint.paper-scroller-background` class.  Attaching a snapshot below -
![image](https://user-images.githubusercontent.com/71628906/194584184-ce45f783-da46-4747-980e-cdb7e1a57c46.png)

Is there a way we adjust the width and height of the paperscroller div so that the paper scales and fits to content?

Also, Is there a way to hide the vertical and horizontal scrollbars on the paperscroller? I have a paper embedded inside a component and I am unfortunately seeing two scrollbars. 
![image](https://user-images.githubusercontent.com/71628906/194585114-195ab654-9a7e-47ba-ad88-8c0b59a76f38.png)

**Comment by jamesgeorgewilliams** [comment, marked as answer]
The `PaperScroller` is a wrapper for the `paper` element.  The `width` and `height` will be set on the paper as you have done above.

The `PaperScroller` changes some transformations (scale, translate). If you are using options like `autoResizePaper`, or methods like `center()` and `fitToContent()`, they will perform some transformations, so the original `width` and `height` you set won't be reflected in dev tools.

If you remove these options and methods as a test from your code, you should observe `100%` and `700px` as you have set them. 

The `paperScroller` option `autoResizePaper: true` will automatically resize the paper so that it fits the content inside it. 

If you are seeing 2 sets of scrollbars, it is probably related to CSS. The `paperScroller` fits the container it is sitting in, and it shouldn't grow to infinity. 

If you have a parent container and child, where the child is the parent of the `paperScroller`, the CSS could be something as follows:

```css
 .parent {
      width: --your-width;
      height: --your-height;
      margin: 0;
      overflow: hidden;
  }
  
  .child {
      top: 0;
      bottom: 0;
      left: 0;
      right: 0;
      position: absolute;
  }  
```
